### DISCUSSION TITLE
terminal rule,why can't it match `TIME` or `DATE`

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1490] [Q&A] [answered]
why can't it match `TIME` or `DATE`,a little weird,lower case `time` or `date` can match correctly,is it because keyword or something?
![image](https://github.com/eclipse-langium/langium/assets/26995040/6e6c3d86-2350-4139-8f3d-a70ba8013b1d)

full langium file is below simple.txt
[simple.txt](https://github.com/eclipse-langium/langium/files/15270060/simple.txt)

PROGRAM main
  VAR
    a:date;
    b:time ;
    c:DATE;
    d:TIME;
    e:Date;
    f:DAte;
  END_VAR

**Comment by zrwsmd** [comment]
write like this seems works well
![image](https://github.com/eclipse-langium/langium/assets/26995040/7a9aea89-8c1c-4459-a47d-e99cef847ec0)

**Comment by msujew** [comment, marked as answer]
Hey @zrwsmd,

I'm a bit confused, the grammar you've linked works well for me, see [here](https://langium.org/playground?grammar=OYJwhgthYgBAygFwFDIKYDtEgJ4MQFyqwkAUADiAPaiQDUAvAArW0SzICUAVMgNyoWNcBCIkA5EwBKAeQDiUgIIBZcbAyQ0DAJIATTIgCWAM0NoQyEmUMZyAV0QBnRttsPHAfQAyhx4g%2BONsAANmg8llawAsiu9k7evv6BGCFosDbG5o6wse4%2BfvBBoUQAkPqmGIZGVBgMACJoFVWGNQDSaDgA7lQgurCkEFS6JmYgjgzKQyPm4SUlpKRVaBDZjLAAajAeDQDGwR6uVZyz4gCiAHJ1HuuKUuL8qJsg22h7B5WEyCUAbjCGYAAjULjPQGaZwUjiRQAFTUdUMIFe-l%2BIH%2BQLQHkojUMAA9OAB%2BWDiAj3EqIHDkNDnTQMalGb4Y6EUjHUiBoB7IBpNaoYdpdHp9RGIOwgDDZPyolJiKziG53WAAHyJcoO5yYAFVYYrlbcPDJNRqtUrZbrtOc9Zq1MaVdDTsomFadVIPKcABq2qTnRReR0m51yLwyABC3vE0Umw1M5lgQpFYtgEqC0okAGEZOd4NDFOcjUSpKcs2bxArxOd0x584XzvdonTDAyPEzKR5WWhSq5EGhgOYPOTmxo2QxCsAMGhdB4bJ3u88%2BxiB2kleqxYYR2OJ1guz3Zx55wwl4FV%2BPJ5uZ8yd5pFUGqFR9tvd1eb72z-OFUhJcAn-2aW%2Bgp%2B55oFQiKwgyqAJsF-O8aRApJwJSP9zzZbUpDQMBb2fGlkNQ%2BD50VICSAAMTsDAdh5DwgSoHYAGtsJpQjiNI8iqJotkFUUEBwBwDxgiWMYOTYjiuJ4xwxChKQlAATTUcQAG01FIUJjEQHQN2nIkADo1LUGMVwACyUjsT36cQABpxB4Tg6CJABdKSZHwtRdDARAwCbLQ6wbVyW00aJkE7EAIBsVDYEwtCvzZAhYAAei8fNvQVWKvEiwwBD8gKNGCBAV1HI8VK3dDwqi%2BAzWhBVioVOoyq8YqkuS3zzDSoL9yytdj2nZi2yi9UipzBV1TK9UKp69UqpzSKUvqwKMugsD33aiLIqDcTbQVAB1GQpDqcq1o2hUvG2zboW0ZRTnKmFTiS8b-Mm2AH1C-8CoWmQZES2rUuun84Mgh7MykM05AuuqrvS2A6JIloMDI4IKOor6OsilNoU2hH1QVfDG1%2BuR4vgeL0e0TH4CkBVoTson0wBwGGoy0EsHBebpI8MAAFoAC9FEZgAtKzpIAHU6DwrO4MbUGQSLIvSfQaajOAdh0mAwBIrIKeu4xwGANksFgJd9DGHYejhjwhbe4GVbANWDFgLwKKc8HMURUwcQiyFtF9ABFX1VE4S7KdgE2zY1wpmYxLF7cd8RXV9INfRW306l9H1PaV4H4WAKphKi6SAAZGYATisuhDYmpOESRDwUTRUJbexB2iQAUjUS2dmtmpK-tzLA5b3F%2BgM1TIU0nJcpAczuC967u-MCLk9T-pSC1rJdcRAlYEnpweGiUXYB0wxdAl5BN%2B3zBYCNoKVvgebuecIW94lw-C6C5QvA8NNlGOnMz8i7nuB5xxufgAX8Q-7mosBBXwPkfDK99H4yGfhcaEHhMxn1IB-L%2BP8-4AM4JfLe18wEIAfk-F%2B0I36AOkgAPW5hgbmIABZC2QEAA&content=AoJQ8g4iCCCyAEBbAhgSwHYCgBq0SfkPmQC4ATZAFwFMCiAjEy1RWo%2BAYxIBFoAVAKJ1CZEnwCSsIe2o8qbIgDMe0GpgEA5bgH1cIIA).
